src/main/java/seedu/address/logic/parser/notes/AddNoteCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser.notes;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.ALL_NOTE_PREFIXES;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TITLE;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.notes.AddNoteCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.notes.note.Description;
+import seedu.address.model.notes.note.Note;
+import seedu.address.model.notes.note.Title;
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddNoteCommandParser implements Parser<AddNoteCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddNOteCommand
+     * and returns an AddNoteCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddNoteCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, ALL_NOTE_PREFIXES);
+
+        if (!arePrefixesPresent(argMultimap, ALL_NOTE_PREFIXES)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddNoteCommand.MESSAGE_USAGE));
+        }
+
+        Title title = ParserUtil.parseTitle(argMultimap.getValue(PREFIX_TITLE).get());
+        Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
+        Note note = new Note(title, description);
+        return new AddNoteCommand(note);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
A lot of commands parsers use this method and it's causing a lot of code duplication. Could you perhaps make this a static `ParserUtil` method so that every command that use prefixes can refer to that?
Namely it would be great to have:
`ParserUtil.arePrefixesPresent()` and `ParserUtil.isAnyPrefixPresent()`

Reply:
Thanks for pointing out the code duplication! I've checked the `ParserUtil` docs and it says "Contains utility methods used for parsing strings in the various Parser classes." so I was just thinking that this would expand the scope of a class already with a lot of methods. If it is better, I've moved it to `Parser` as a default implementation instead!